### PR TITLE
creates new logout feature for student and teacher users

### PR DESCRIPTION
**Pull request by violatido**
This adds logout capabilities for both teachers and students. For now I only have logout buttons on the home page for users, but I might put them on the teacher notes and charts pages.

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -66,6 +66,16 @@ def add_teacher():
     else: 
         return jsonify({'teacher_fname': teacher_fname, 'teacher_lname': teacher_lname})
 
+@app.route('/teacher-logout') 
+def teacher_logout(): 
+    print(session)
```
Comment (collingreen) [inline server.py:71]: probably left this debug print in by accident - should either remove it (or, if you really want the entire session in your logs, sanitize it and add some more informative text )

File: server.py [hunk 2]
```diff
@@ -388,6 +406,8 @@ def send_message():
     return jsonify({'message_content': text_message_content})
 
 
+
+
```
Comment (collingreen) [inline server.py:409]: unrelated whitespace added - may want to clean it up if you're OCD about your commits staying hyper clean

File: server.py [hunk 3]
```diff
@@ -118,6 +128,14 @@ def add_student():
     else: 
         return jsonify({'status': 'error, registration credentials incorrect'})
 
+@app.route('/student-logout')
```
Comment (collingreen) [inline server.py:131]: I don't see why you need two different logout methods/routes -- why not just make one that removes both student_id and teacher_id if they are there?

File: server.py [hunk 4]
```diff
@@ -66,6 +66,16 @@ def add_teacher():
...
         return jsonify({'teacher_fname': teacher_fname, 'teacher_lname': teacher_lname})
 
+@app.route('/teacher-logout') 
+def teacher_logout(): 
+    print(session) 
+    if session['teacher_id']: 
+        session.pop('teacher_id') 
+        return redirect('/') 
+    else: 
+        session.clear()
```
Comment (collingreen) [inline server.py:76]: why are we clearing the session here but not in the case where we actually find the teacher id?

File: server.py [hunk 5]
```diff
@@ -118,6 +128,14 @@ def add_student():
...
         return jsonify({'status': 'error, registration credentials incorrect'})
 
+@app.route('/student-logout') 
+def student_logout(): 
+    print(session) 
+    if session['student_id']: 
+        session.pop('student_id') 
+        return redirect('/') 
+    else: 
+        session.clear()
```
Comment (collingreen) [inline server.py:138]: probably want 2 blank lines after this to match the rest of your file

File: templates/student-profile.html [hunk 6]
```diff
@@ -8,8 +8,9 @@
 <body class="white">
   <nav class="light-pink">
     <div class="row nav-row">
-        <div class="col-12 d-flex justify-content-start left-nav-col">
+        <div class="col-12 d-flex justify-content-between left-nav-col">
           <a class="arch btn btn-lg home-btn-2" href="/">Virtual Music Studio</a>
+          <a class="arch btn btn-sm " href="/student-logout">Logout</a>
```
Comment (collingreen) [inline templates/student-profile.html:13]: nit: extra whitespace in the anchor class

File: requirements.txt [hunk 7]
```diff
@@ -12,7 +12,7 @@ pkg-resources==0.0.0
 psycopg2-binary==2.8.5
 pytz==2021.1
 SQLAlchemy==1.3.17
-pip install twilio==6.52.0
+twilio==6.53.0
```
Comment (collingreen) [inline requirements.txt:15]: this commit doesnt seem related to the logout functionality in this PR - I'd move it to a separate PR about bumping the twilio version

File: server.py [hunk 8]
```diff
@@ -405,8 +405,7 @@ def send_message():
 
     return jsonify({'message_content': text_message_content})
 
-
-
+#_________________________________________________________________________________________________#
```
Comment (collingreen) [inline server.py:408]: these two changes to this file seem like they are just cosmetic - I'm annoying with my commits so I'd put them in their own commit instead of making it seem like they are somehow part of the twilio change

File: server.py (hunk 1 above)
Comment (violatido) [reply server.py:71]: Yes, definitely was left by a mistake! I'm continuing to clean up all my code, including these prints and console.logs!

File: server.py (hunk 2 above)
Comment (violatido) [reply server.py:409]: Good point! Even the commits need to have proper syntax in addition to thoughtful wording.

File: server.py (hunk 3 above)
Comment (violatido) [reply server.py:131]: Good point! I formed a lot of these routes without understand the potential for redundancy and wasteful code. I'll try tidying up the routes ASAP!

File: server.py (hunk 4 above)
Comment (violatido) [reply server.py:76]: From the StackOverflow post I read, session.pop() will remove the variable from the session and you don't have to keep updating your secret key. Would you recommend using session.clear?

File: requirements.txt (hunk 7 above)
Comment (violatido) [reply requirements.txt:15]: I'll try to keep changes in each PR related to singular issues!

File: server.py (hunk 8 above)
Comment (violatido) [reply server.py:408]: I'll continue to enforce strategic commits. While I made frequent commits, I often would make several unrelated changes per commit. I'll work on that!

File: server.py (hunk 4 above)
Comment (violatido) [reply server.py:76]: For now I'm going to keep this comment and the one about routes open in this PR, but I'm going to go ahead and merge since I have resolved the other comments.
